plugin.d.ts

```diff
@@ -41,7 +41,7 @@ export namespace SharedWorker {
 		readonly file: string;
 		publish: (data: Data) => PublishedMessage<Data>;
 		subscribe: () => AsyncIterableIterator<ReceivedMessage<Data>>;
-		teardown: <TeardownFn extends () => void> (fn: TeardownFn) => TeardownFn;
+		teardown: (fn: (() => Promise<void>) | (() => void)) => () => Promise<void>;
```

Comment:
```suggestion
		teardown: (fn: (() => Promise<void> | void) => () => Promise<void>;
```

?

Reply:
Would that be fine or would `void` "nullify" `Promise<void>`? Or would it be clearer to have overloads?

Reply:
Hmm: https://github.com/microsoft/TypeScript/issues/43921

I guess what you have is fine.

Reply:
I don't really know if overloads will make any difference.